### ISSUE TITLE
Broken scraper: opinions.united_states.federal_special.tax

### ISSUE THREAD

**Issue by arderyp** [freelawproject/juriscraper#303]
```
  Traceback (most recent call last):
    File "./arderyp_test.py", line 470, in run_scraper
      scraper.parse()
    File "/home/arderyp/code/git/juriscraper/juriscraper/AbstractSite.py", line 123, in parse
      self.html = self._download()
    File "/home/arderyp/code/git/juriscraper/juriscraper/opinions/united_states/federal_special/tax.py", line 57, in _download
      start_date_input = driver.find_element_by_id(start_date_id)
    File "/home/arderyp/.virtualenvs/juriscraper/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 269, in find_element_by_id
      return self.find_element(by=By.ID, value=id_)
    File "/home/arderyp/.virtualenvs/juriscraper/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 752, in find_element
      'value': value})['value']
    File "/home/arderyp/.virtualenvs/juriscraper/local/lib/python2.7/site-packages/selenium/webdriver/remote/webdriver.py", line 236, in execute
      self.error_handler.check_response(response)
    File "/home/arderyp/.virtualenvs/juriscraper/local/lib/python2.7/site-packages/selenium/webdriver/remote/errorhandler.py", line 192, in check_response
      raise exception_class(message, screen, stacktrace)
  NoSuchElementException: Message: {"errorMessage":"Unable to find element with id 'ctl00_Content_dpDateSearch_dateInput'","request":{"headers":{"Accept":"application/json","Accept-Encoding":"identity","Connection":"close","Content-Length":"117","Content-Type":"application/json;charset=UTF-8","Host":"127.0.0.1:52955","User-Agent":"Python-urllib/2.7"},"httpVersion":"1.1","method":"POST","post":"{\"using\": \"id\", \"sessionId\": \"48693b40-5ac3-11ea-bde4-2fd95526c968\", \"value\": \"ctl00_Content_dpDateSearch_dateInput\"}","url":"/element","urlParsed":{"anchor":"","query":"","file":"element","directory":"/","path":"/element","relative":"/element","port":"","host":"","password":"","user":"","userInfo":"","authority":"","protocol":"","source":"/element","queryKey":{},"chunks":["element"]},"urlOriginal":"/session/48693b40-5ac3-11ea-bde4-2fd95526c968/element"}}
  Screenshot: available via screen
```

This scraper is failing because PhantomJS seems to have a problem rendering HTTPS urls.  All our other PhantomJS scrapers are HTTP, and they work.  I've done some testing and I can render/screenshot HTTP urls via the PhantomJS webdriver, but not various HTTP urls.  This is the same problem that's currently blocking a fix to #296.

@mlissner, I was surprised to discover this scraper is failing.  I only discovered it because I was testing the PhantonJS HTTPS problem when I discovered it while trying to address #296.  I was surprised because I haven't seen this scraper mentioned in any of the log reports.  I double checked and still don't see it there.  I also checked the CL log interface and don't see anything.  In fact, the CL logs seems to indicate this scraper isn't even running.  Do you by chance have more context here that I may be missing?

**Comment by arderyp** [comment]
@mlissner, the common solution for this problem that gets cited is this: https://stackoverflow.com/questions/23581291/python-selenium-with-phantomjs-empty-page-source/34257642#34257642

That doesn't work with phantomjs_executable_path, which points to phantomjs-1.9.7 on my machine.  It, if I swap the config to use phantomjs-2.1.1 instead, everything works.  Do you know which versions our servers use?

**Comment by flooie** [comment]
This is definitely my fault.  I think I have a solution though.  Can I remove the selenium?

**Comment by arderyp** [comment]
I'm not sure i understand.  What is you fault? And remove selenium form what?  The juriscraper server, or the selenium code from this scraper... or?

**Comment by flooie** [comment]
@arderyp I dont think we need to use selenium for Tax.  I rewrote the scraper to rely only upon requests.  Although I'm still a little rough around the edges its tests correctly on my machine.  I would love to get your thoughts.

https://github.com/freelawproject/juriscraper/blob/new-tax/juriscraper/opinions/united_states/federal_special/tax.py

**Comment by arderyp** [comment]
awesome!  I will take a look tomorrow.  But I need @mlissner 's feedback anyways on the PhantonJS version that we depend on, since my testing suggests I need 2.1.1+ in order to fix #296.  Also, I'm interested to see if @mlissner knows why this scraper doesn't appear to be running at at in production.

**Comment by arderyp** [comment]
well @flooie, I guess you are reworking kan too to not use selenium either.  I do wonder if a selenium approach might be more intuitive, but if your changes are working then go with it.  Selenium isn't the most intuitive mechanism either.

**Comment by arderyp** [comment]
hey @flooie, I know I mentioned going with your non-selenium fix for kan, but I think I've got an implementation thats a little more intuitive than what I see on your "fix-kansas" branch.  Nothing against your approach, its just quite difficult to mentally parse the nonsense required to manually request through an ASPX app (hence my disdain for working with such pages).  I think clicking through with selenium is a little more intuitive and therefor maintainable.

But, @mlissner, my fix (which breaks kan and kanctapp into kan_p, kan_u, kanctapp_p, kanctapp_u) requires a non-ancient version of PhantonJS.  Regardless of whether we go with my fix, we should get off PhantonJS 1.9.7 anyways as its unable to parse HTTPS pages, and that's something we definitely need to be able to do.

Please let me know @mlissner.

**Comment by arderyp** [comment]
this is no longer broken... upgrade of PhantonJS (https://github.com/freelawproject/juriscraper/commit/f1b32a6959ef78eaa59dc3e2f8a1a09ba5f8813c) seems to have done the trick
